src/core/qfieldcloudprojectsmodel.cpp

```diff
@@ -1033,24 +1033,24 @@ void QFieldCloudProjectsModel::projectDownload( const QString &projectId )
     const QJsonArray files = payload.value( QStringLiteral( "files" ) ).toArray();
     for ( const QJsonValue &fileValue : files )
     {
-      QJsonObject fileObject = fileValue.toObject();
-      int fileSize = fileObject.value( QStringLiteral( "size" ) ).toInt();
-      QString fileName = fileObject.value( QStringLiteral( "name" ) ).toString();
-      QString projectFileName = QStringLiteral( "%1/%2/%3/%4" ).arg( QFieldCloudUtils::localCloudDirectory(), mUsername, projectId, fileName );
-      QString cloudChecksumMd5 = fileObject.value( QStringLiteral( "md5sum" ) ).toString();
-      QString localChecksumMd5 = FileUtils::fileChecksum( projectFileName, QCryptographicHash::Md5 ).toHex();
+      const QJsonObject fileObject = fileValue.toObject();
+      const int fileSize = fileObject.value( QStringLiteral( "size" ) ).toInt();
+      const QString fileName = fileObject.value( QStringLiteral( "name" ) ).toString();
+      const QString projectFileName = QStringLiteral( "%1/%2/%3/%4" ).arg( QFieldCloudUtils::localCloudDirectory(), mUsername, projectId, fileName );
+      const QString cloudChecksum = fileObject.value( QStringLiteral( "md5sum" ) ).toString();
+      const QString localChecksum = FileUtils::fileEtag( projectFileName );
```

Comment:
```suggestion
      // NOTE the cloud API is giving the false impression that the file keys `md5sum` is having a MD5 or another checksum. This is actually an implementation specific Object Storage `E-Tag`.
      const QString cloudEtag = fileObject.value( QStringLiteral( "md5sum" ) ).toString();
      const QString localEtag = FileUtils::fileEtag( projectFileName );
```

Just to be explicit about the confusion I introduced myself two years ago.